### PR TITLE
CLI: support multiple files as input

### PR DESCRIPTION
**Pull request by 2010YOUY01**
Signed-off-by: Yongting You <2010youy01@gmail.com>

Close [#117 ](https://github.com/risinglightdb/sqllogictest-rs/issues/117)
This PR adds multiple blob files CLI support (as #117 suggested)

The following CLI inputs have been tested locally
1. One unquoted glob
```
> ./target/debug/sqllogictest ./**/*.slt 
  ...(tests runned)
```
2. No input file
```
> ./target/debug/sqllogictest
  error: The following required arguments were not provided:
  <FILES>...

   USAGE:
   sqllogictest [OPTIONS] <FILES>...

   For more information try --help
```
3. Multiple globs
```
> ./target/debug/sqllogictest ./examples/basic/*.slt "./examples/condition/*.slt" ./examples/rowsort/rowsort.slt
    ...(tests runned)
```

### CODE REVIEW

File: sqllogictest-bin/src/lib.rs [hunk 1]
```diff
@@ -170,8 +170,16 @@ pub async fn main_okk() -> Result<()> {
...
     }
 
-    let files = glob::glob(&files).context("failed to read glob pattern")?;
-    let files = files.into_iter().try_collect::<_, Vec<_>, _>()?;
+    let files: Vec<PathBuf> = files
+        .iter()
+        .map(|filename| {
+            glob::glob(filename)
+                .context("failed to read glob pattern")
+                .unwrap()
```
Comment (skyzh) [inline sqllogictest-bin/src/lib.rs:178]: Can we return error instead of unwrap here?

**Review by skyzh — COMMENTED** [review]
Rest LGTM, good work!

File: sqllogictest-bin/src/lib.rs (hunk 1 above)
Comment (skyzh) [reply sqllogictest-bin/src/lib.rs:178]: ... and also the `unwrap` below

Comment (2010YOUY01) [reply sqllogictest-bin/src/lib.rs]: Fixed, thank you for your advice!

File: sqllogictest-bin/src/lib.rs [hunk 2]
```diff
@@ -170,8 +170,17 @@ pub async fn main_okk() -> Result<()> {
...
+        .into_iter()
+        .map(|filename| glob::glob(&filename).context("failed to read glob pattern"))
+        .try_collect::<_, Vec<_>, _>()?
+        .into_iter()
+        .map(|glob| glob.try_collect::<_, Vec<_>, _>())
+        .try_collect::<_, Vec<_>, _>()?
+        .into_iter()
+        .flatten()
+        .collect();
+
```
Comment (xxchan) [inline sqllogictest-bin/src/lib.rs:183]: The repeated `try_collect` and `into_iter` don't look very nice... Maybe simple for loops are just more ergonomic in this case.

Comment (xxchan) [reply sqllogictest-bin/src/lib.rs:183]: Alternatively you can use `and_then` in the first `map`. In this way you can save on `try_collect`

Comment (skyzh) [reply sqllogictest-bin/src/lib.rs:183]: Actually I feel for loop might be more expressive…

Comment (2010YOUY01) [reply sqllogictest-bin/src/lib.rs]: I got brainwashed by the pipe style previously, changed it to `for` and it looks much better.
Much appreciated for the feedback!

**Comment by skyzh** [comment]
Just some clippy warnings and we are ready to merge!

**Comment by 2010YOUY01** [comment]
> Just some clippy warnings and we are ready to merge!

clippy is complaining about one formatting problem in multiple files I didn't change
```
26 -             write!(f, "\nat {}", upper)?;
26 +             write!(f, "\nat {upper}")?;
```
`cargo clippy --fix` seems to work, should I add a commit to fix these warnings?

**Comment by skyzh** [comment]
Yes, thanks a lot! Looks like due to the recent Rust stable upgrade.

**Review by skyzh — APPROVED** [review]

**Pull request merged**